### PR TITLE
Added a check for active tool

### LINKED ISSUE #1688: Drawing Tools: Interaction Layer shouldn't run onPointerDown when there's no drawing task
## General Bug

Package: lib-classifier
Initially discussed in #1677

If the Classifier is in Annotate mode, but there's no active drawing task (or no active drawing tool selected, presumably), then clicking on the Subject will cause a hidden `Uncaught TypeError: activeTool.createMark is not a function, at onPointerDown` error in the dev console.

Note: this is a silent error and doesn't crash anything.

Note: In #1677, this was misidentified as _"clicking on an existing, inert drawing mark causes a hidden Uncaught TypeError: activeTool.createMark is not a function, at onPointerDown error in the dev console"_

### Testing

Setup:
- Create a workflow with a non-drawing task. (e.g. a single answer question)

Actions:
- Make the Classifier enter Annotate mode (click on the Annotate arrow on the central image toolbar)
- Click anywhere on the Subject image in an attempt to "annotate" it with a drawing mark (using a drawing tool that doesn't exist).
- Expected: nothing should happen
- Actual: a hidden error message appears in the dev console.

Test on `http://localhost:8080/?env=staging&project=1898&workflow=3382` on macOS 10 + Chrome 83

### Recommendation

Just a guess: in `src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js`, in `onPointerDown()`...

- The line `const activeMark = activeTool.createMark({ ... })` doesn't check if the current active tool exists.
- I assume a simple `if (activeTool && activeTool.createMark) ...` check should solve the problem, but I may be missing something. It's too simple.

### Status

Low priority (nothing is crashing) but important to sort out.

### PR DESCRIPTION
**Pull request by ErikOstlund**
Package: `lib-classifier`
Fixes #1688 

Describe your changes:
`onPointerDown` method now checks for an active tool before running. Several prettier updates.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
-
-const StyledRect = styled('rect')`
-  ${props => props.disabled ? 
-    css`cursor: not-allowed;` :
-    css`cursor: crosshair;`
-  }
-`
-
-function InteractionLayer ({
+import cuid from "cuid";
```
Comment (shaunanoordin) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js:1]: Style guide: `prettier` is adding semicolons to a lot of these lines; these aren't required.

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js [hunk 2]
```diff
@@ -27,105 +31,105 @@ function InteractionLayer ({
...
-      id: cuid(),
-      frame,
-      toolIndex: activeToolIndex
-    })
-    activeMark.initialPosition(convertEvent(event))
-    setActiveMark(activeMark)
-    setCreating(true)
-    activeMark.setSubTaskVisibility(false)
-    return false
+    console.log("activeTool: ", activeTool);
```
Comment (shaunanoordin) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js:65]: `console.log()` needs to be removed

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.spec.js [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
+import sinon from "sinon";
+
+import SVGContext from "@plugins/drawingTools/shared/SVGContext";
+import InteractionLayer, { StyledRect } from "./InteractionLayer";
+import TranscribedLines from "./components/TranscribedLines";
+import SubTaskPopup from "./components/SubTaskPopup";
+import DrawingTask from "@plugins/tasks/DrawingTask";
+import { Line, Point } from "@plugins/drawingTools/components";
+
+describe.only("Component > InteractionLayer", function() {
```
Comment (shaunanoordin) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.spec.js:12]: This should be `describe()`

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js [hunk 4]
```diff
@@ -27,105 +31,105 @@ function InteractionLayer ({
...
+        id: cuid(),
+        frame,
+        toolIndex: activeToolIndex
+      });
+      activeMark.initialPosition(convertEvent(event));
+      setActiveMark(activeMark);
+      setCreating(true);
+      activeMark.setSubTaskVisibility(false);
+      return false;
+    }
```
Comment (shaunanoordin) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js:78]: This if-block works great in making sure invalid `activeTool` values don't screw up the app.

However, an easier alternative might be to use `if (!activeTool.type) return false` and not have an if-block

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js [hunk 5]
```diff
@@ -27,105 +31,105 @@ function InteractionLayer ({
...
+    return;
```
Comment (shaunanoordin) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js:79]: Recommend: if we're using an if-block, this should be `return false` to be consistent with original code

**Comment by ErikOstlund** [comment]
Reverted original commit because my global prettier rules were adding semicolons everywhere and made a huge mess!
In the latest commit,
prettier is now following Standard JS style guide rules.
Removed the console.log
Updated `describe.only()` to `describe()`
Updated `activeTool.type` check as recommended for code clarity and consistency.
All unit tests are passing.

**Pull request merged**